packages/web/src/generators/static-serve/static-serve-configuration.ts

```diff
@@ -1,42 +1,54 @@
 import {
+  createProjectGraphAsync,
   logger,
   parseTargetString,
+  type ProjectGraph,
+  readCachedProjectGraph,
   readProjectConfiguration,
   stripIndents,
-  TargetConfiguration,
-  Tree,
+  type TargetConfiguration,
+  type Tree,
   updateProjectConfiguration,
 } from '@nx/devkit';
-import { Schema as FileServerExecutorSchema } from '../../executors/file-server/schema.d';
+import type { Schema as FileServerExecutorSchema } from '../../executors/file-server/schema.d';
 
 interface WebStaticServeSchema {
   buildTarget: string;
   outputPath?: string;
   targetName?: string;
+  spa?: boolean;
 }
 
 interface NormalizedWebStaticServeSchema extends WebStaticServeSchema {
   projectName: string;
   targetName: string;
+  spa: boolean;
 }
 
-export function webStaticServeGenerator(
+export async function webStaticServeGenerator(
   tree: Tree,
   options: WebStaticServeSchema
 ) {
-  const opts = normalizeOptions(tree, options);
+  const opts = await normalizeOptions(tree, options);
   addStaticConfig(tree, opts);
 }
 
-function normalizeOptions(
+async function normalizeOptions(
   tree: Tree,
   options: WebStaticServeSchema
-): NormalizedWebStaticServeSchema {
-  const target = parseTargetString(options.buildTarget);
+): Promise<NormalizedWebStaticServeSchema> {
+  let projectGraph: ProjectGraph;
+  try {
+    projectGraph = readCachedProjectGraph();
+  } catch (e) {
+    projectGraph = await createProjectGraphAsync();
+  }
```

Comment:
We need to be careful about using the project graph in generators. If a project generator (e.g. `@nx/next:app`) invokes this generator, the project being generated is not yet in the project graph because the generation changes have not been flushed to disk yet. So, anything that relies on the project graph to get the still-in-memory project wouldn't find it.

I'm unsure if there's indeed a problem here or not, but we definitely need to check all usages of this generator in other generators and validate the above.

Reply:
It is only being used by `parseTargetString` which handles the case of when the project graph does not contain the project. 

But for cases when the generator is invoked via CLI, it is more accurate.